google_guest_agent/accounts_unix.go

```diff
@@ -23,20 +23,28 @@ import (
 	"syscall"
 )
 
-func getUID(path string) string {
+func getUIDAndGID(path string) (string, string) {
 	if dir, err := os.Stat(path); err == nil {
 		if stat, ok := dir.Sys().(*syscall.Stat_t); ok {
-			return fmt.Sprintf("%d", stat.Uid)
+			return fmt.Sprintf("%d", stat.Uid), fmt.Sprintf("%d", stat.Gid)
 		}
 	}
-	return ""
+	return "", ""
 }
 
-func createUser(username, uid string) error {
+func createUser(username, uid, gid string) error {
 	useradd := config.Section("Accounts").Key("useradd_cmd").MustString("useradd -m -s /bin/bash -p * {user}")
 	if uid != "" {
 		useradd = fmt.Sprintf("%s -u %s", useradd, uid)
 	}
+	if gid != "" {
```

Comment:
You want to first add the group then the user, don't you? If we create a user first its group will be also created and then if you later to add the group the group will already exist resulting into a error i.e. `groupadd: group '<user>' already exists`.

Reply:
I misread the change in the first pass. It's indeed creating the group first.

Reply:
I'd rewrite this code to something like this:

```golang
func createUser(username, uid string, gid string) error {
       if gid != "" {
		groupadd := config.Section("Accounts").Key("groupadd_cmd").MustString("groupadd {group}")
		groupadd = fmt.Sprintf("%s -g %s", groupadd, gid)
		if err := runCmd(createUserGroupCmd(groupadd, "", username)); err != nil {
			return err
		}
       }

        useradd := config.Section("Accounts").Key("useradd_cmd").MustString("useradd -m -s /bin/bash -p * {user}")
        if uid != "" {
                useradd = fmt.Sprintf("%s -u %s", useradd, uid)
        }
        return runCmd(createUserGroupCmd(useradd, username, ""))
}
```